Remove lodash pick

AndrewMusgrave opened the pull request:
### WHY are these changes introduced?

Part of removing lodash #991 

### WHAT is this pull request doing?

Creating a `pick` function and using that instead of `_.pick`

### How to 🎩
Test pass? 💯 

### TODO:
* add changelog

tmlayton commented on src/utilities/pick.ts:
```diff
@@ -0,0 +1,8 @@
+
+function pick(obj: GeneralObject | null, keyPaths: string[]) {
+  if (obj == null || keyPaths.length === 0) return {};
+  return keyPaths.reduce((acc, key) => ({...acc, [key]: obj[key]}), {});
```
If a key path is passed that doesn’t exist in the object, it will return the key with value `undefined`. Is that your intent? Either way, I would add a test case for the expectation if a key path is not present in the object.

tmlayton commented on src/utilities/tests/pick.test.ts:
```diff
@@ -0,0 +1,23 @@
+  });
+
+  it('returns a object with values picked from the provided argument', () => {
+    expect(pick({propA: 1, propB: 2, propC: 3}, ['propA', 'propB'])).toEqual({
```
```suggestion
    expect(pick({propA: 1, propB: 2, propC: 3}, ['propA', 'propC'])).toEqual({
```

tmlayton commented on src/utilities/tests/pick.test.ts:
```diff
@@ -0,0 +1,23 @@
+  it('returns a object with values picked from the provided argument', () => {
+    expect(pick({propA: 1, propB: 2, propC: 3}, ['propA', 'propB'])).toEqual({
+      propA: 1,
+      propB: 2,
```
```suggestion
      propC: 3,
```

tmlayton commented:
I’m in favor of this being a simpler, but less feature-rich implementation of lodash pick because we don’t use most of what the lodash pick version supports. But for comparison, here is the lodash test suite for pick running with this version https://codepen.io/timlayton/pen/MLdNrv.

The only thing I could see worth adding is deeply nested picking, but I’m good with shallow picking for now.

AndrewMusgrave commented:
Sorry addressing feedback took so long, still finishing / catching up from my tour 🙇 I added nested picking and don't include unmatched keypaths. Let me know what you think or if you have any suggestions!
